src/swupd/manifest.go

```diff
@@ -0,0 +1,118 @@
+package swupd
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// ManifestHeader contains metadata for the manifest
+type ManifestHeader struct {
+	Format      int
+	Version     int32
+	Previous    int32
+	FileCount   int32
+	TimeStamp   time.Time
+	ContentSize int64
+	Includes    []Manifest
+}
+
+// Manifest represents a bundle or list of bundles (MoM)
+type Manifest struct {
+	Name         string
+	Header       ManifestHeader
+	Files        []File
+	AddedCount   uint32
+	RemovedCount uint32
+	ChangedCount uint32
+}
+
+// MoM is a manifest of manifests with the same header information
+type MoM struct {
+	Header       ManifestHeader
+	SubManifests []Manifest
+}
+
+// readManifestFileHeaderLine Read a header line from a manifest
+func readManifestFileHeaderLine(fields []string, m *Manifest) error {
+	var err error
+	var parsed uint64
+
+	// Only search for defined fields
+	switch fields[0] {
+	case "MANIFEST":
```

Comment:
Question: does the manifest format assumes that MANIFEST appears only once and only in the first line? (It seems to me useful to assume that from the format)

Reply:
The "Only" once thing might apply to other properties too.

Because of those properties that span multiple line (e.g. headers that must be unique, who needs to appear first), would be useful to later on just inline this function into the actual caller code parsing the header.